src/rat/plugins/forecasting/forecasting.py

```diff
@@ -714,38 +713,49 @@ def forecast_outflow(
 
 def generate_forecast_state_and_inputs(
         forecast_startdate, # forecast start date
-        forecast_enddate, # forecast end date
+        forcings_enddate, # forecast end date
         hindcast_combined_datapath, 
         forecast_combined_datapath, 
-        out_dir
+        out_dir,
+        forcings_startdate = None
     ):
+    # If metsim start date is no provided, then start from forecast start date
+    if not forcings_startdate:
+        forcings_startdate = forecast_startdate
```

Comment:
this line initializes the forcings_startdate variable. in line 747 this will lead to running of the else statement always.

Reply:
Drafted a new generalized function in metsim_input_processing.py